File: WASMXamarin/WASMXamarin/WASMXamarin.Droid/MobileArrayAdapter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Android.App;
using Android.Content;
using Android.Graphics;
using Android.OS;
using Android.Runtime;
using Android.Views;
using Android.Widget;
using Android.Media;

namespace WASMXamarin.Droid
{
    public class MobileArrayAdapter : ArrayAdapter<String>, TextureView.ISurfaceTextureListener
    {
        private Context context;
        private String[] values;

        public MobileArrayAdapter(Context context, string[] values) : base(context, Resource.Layout.activity_layout, values)
        {
            this.context = context;
            this.values = values;
        }

        public override View GetView(int position, View convertView, ViewGroup parent)
        {
            LayoutInflater inflater = (LayoutInflater)context.GetSystemService(Context.LayoutInflaterService);

            View rowView = inflater.Inflate(Resource.Layout.activity_layout, parent, false);

            TextureView textureView = rowView.FindViewById<TextureView>(Resource.Id.textureView);
            textureView.SurfaceTextureListener = this;

            return rowView;
        }

        public void OnSurfaceTextureAvailable(SurfaceTexture surface, int width, int height)
        {
            MediaPlayer mediaPlayer = MediaPlayer.Create(context.ApplicationContext, context.Resources.GetIdentifier("video", "raw", context.PackageName));
            mediaPlayer.SetSurface(new Surface(surface));
            try
            {
                mediaPlayer.Start();
                mediaPlayer.Looping = true;
            }
            catch (Exception e)
            {
            }
        }

        public bool OnSurfaceTextureDestroyed(SurfaceTexture surface)
        {
            return true;
        }

        public void OnSurfaceTextureSizeChanged(SurfaceTexture surface, int width, int height)
        {
        }

        public void OnSurfaceTextureUpdated(SurfaceTexture surface)
        {
        }
    }
}

File: Native.Android/app/src/main/java/com/example/konra/thespanishinquisition/MobileArrayAdapter.java
package com.example.konra.thespanishinquisition;

import android.app.Activity;
import android.app.ProgressDialog;
import android.content.Context;
import android.content.res.Configuration;
import android.graphics.SurfaceTexture;
import android.media.MediaPlayer;
import android.media.MediaPlayer.OnPreparedListener;
import android.net.Uri;
import android.os.Bundle;
import android.provider.MediaStore;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.Surface;
import android.view.TextureView;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.MediaController;
import android.widget.VideoView;

public class MobileArrayAdapter extends ArrayAdapter<String> implements TextureView.SurfaceTextureListener {
    private final Context context;
    private final String[] values;

    public MobileArrayAdapter(Context context, String[] values) {
        super(context, R.layout.activity_layout, values);
        this.context = context;
        this.values = values;
    }
    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        LayoutInflater inflater = (LayoutInflater) context
                .getSystemService(Context.LAYOUT_INFLATER_SERVICE);

        View rowView = inflater.inflate(R.layout.activity_layout, parent, false);

        TextureView textureView = (TextureView)rowView.findViewById(R.id.textureView);
        textureView.setSurfaceTextureListener(this);

        return rowView;
    }

    @Override
    public void onSurfaceTextureAvailable(SurfaceTexture surface, int width, int height) {
        MediaPlayer mediaPlayer = MediaPlayer.create(context.getApplicationContext(), context.getResources().getIdentifier("video", "raw", context.getPackageName()));
        mediaPlayer.setSurface(new Surface(surface));
        try {

            mediaPlayer.start();
            mediaPlayer.setLooping(true);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    @Override
    public void onSurfaceTextureSizeChanged(SurfaceTexture surface, int width, int height) {

    }

    @Override
    public boolean onSurfaceTextureDestroyed(SurfaceTexture surface) {
        return false;
    }

    @Override
    public void onSurfaceTextureUpdated(SurfaceTexture surface) {

    }
}